packages/scandipwa/src/util/Promise/PauseUntil.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/scandipwa
+ */
+
+export const CHECK_INTERVAL = 100;
+export const WAIT_TIMEOUT = 10000;
+
+/** @namespace Util/Promise/PauseUntil/pauseUntil */
+export async function pauseUntil(condition, error, timeoutMs = WAIT_TIMEOUT, checkInterval = CHECK_INTERVAL) {
+    return new Promise((resolve, reject) => {
+        const startTime = new Date();
+        const wait = setInterval(() => {
+            if (condition()) {
+                clearInterval(wait);
+                resolve('Wait successful');
+            }
+
+            if (new Date() - startTime > timeoutMs) {
+                clearInterval(wait);
+                reject(new Error(error || 'Wait timeout'));
+            }
+        }, checkInterval);
+    });
+}
```

Comment:
It is understandable why this logic is implemented, but in my opinion it is not good practice to use timers for checking something at all. It will be shooting each 0.1 sec which is not cool tho. Maybe we need to implement some kind of block from adding to whishlist (loader) while it is updating ? In default magento it is all can be updated only if resave product in whislist from PDP with correct qty (can't say that it is more user friendly), but when adding to cart it gets qty from form.  Maybe we need to refactor wishlist logic or add block from while updating it ?